readthedocs/projects/views/base.py

```diff
@@ -115,3 +115,17 @@ def post(self, request, *args, **kwargs):
 
     def get_failure_url(self):
         return reverse('homepage')
+
+
+class Echo:
+
+    """
+    A class that implements just the write method of the file-like interface.
+
+    This class can be used for generating StreamingHttpResponse.
+    See: https://docs.djangoproject.com/en/2.2/howto/outputting-csv/#streaming-large-csv-files
+    """
+
+    def write(self, value):
+        """Write the value by returning it, instead of storing in a buffer."""
+        return value
```

Comment:
I'd probably put this in a `utils.py` class, not a base View.

Reply:
Done. :+1: